src/database/schema/UserSchema.ts

```diff
@@ -224,6 +224,23 @@ class UserSchema implements BaseDbSchema {
     }
   };
 
+  static async deleteByUserId(
+    db: SQLiteDatabase,
+    userId: string,
+    channelId: string
+  ): Promise<void> {
+    const resolver = (tx: Transaction) => {
+      const query = `DELETE FROM ${UserSchema.getTableName()} WHERE user_id = ? AND channel_id = ? LIMIT 1`;
+      tx.executeSql(query, [userId, channelId]);
+    };
+
+    try {
+      db.transaction(resolver);
+    } catch (e) {
+      console.log('error resolver', e);
+    }
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `deleteByUserId` function is using a transaction to execute the SQL query. However, it's not handling the promise returned by the `transaction` method. The `transaction` method returns a Promise that resolves when the transaction has been completed. You should await this promise to ensure that the transaction completes before continuing execution.

Also, the error handling here could be improved. Currently, if an error occurs during the transaction, it's logged to the console and then ignored. This could make debugging difficult if something goes wrong. It would be better to throw the error after logging it, so that it can be caught and handled by the caller.

Here's how you might modify the code:

```diff
    try {
-       db.transaction(resolver);
+       await db.transaction(resolver);
    } catch (e) {
      console.log('error resolver', e);
+       throw e;
    }
```
This way, the function will wait for the transaction to complete before returning, and any errors that occur will be propagated up to the caller.


<!-- This is an auto-generated comment by OSS CodeRabbit -->